lib/logux/class_finder.rb

```diff
@@ -2,43 +2,50 @@
 
 module Logux
   class ClassFinder
-    attr_reader :params
+    attr_reader :action, :meta
 
-    def initialize(params)
-      @params = params
+    def initialize(action:, meta:)
+      @action = action
+      @meta = meta
     end
 
     def find_action_class
       "#{class_namespace}::#{class_name}".constantize
     rescue NameError
-      raise Logux::NoActionError, %(
+      message = %(
         Unable to find action #{class_name.camelize}
-        Should be in app/logux/actions/#{class_path}.rb
+        Should be in app/logux/#{action_type}s/#{class_path}.rb
       )
+      raise Logux::NoActionError.new(message, meta: meta) if action_type == :action
+      raise Logux::NoChannelError.new(message, meta: meta)
     end
 
     def find_policy_class
       "Policies::#{class_namespace}::#{class_name}".constantize
     rescue NameError
-      raise Logux::NoPolicyError, %(
+      raise Logux::NoPolicyError.new(%(
         Unable to find policy #{class_name.camelize}
         Should be in app/logux/policies/#{class_path}.rb
-      )
+      ), meta: meta)
     end
 
     def class_namespace
-      return 'Channels' if params.type == 'logux/subscribe'
+      return 'Channels' if action_type == :channel
       'Actions'
     end
 
     def class_name
-      if params.type == 'logux/subscribe'
-        params.channel_name.camelize
+      if action_type == :channel
+        action.channel_name.camelize
       else
-        params.type.split('/')[0..-2].map(&:camelize).join('::')
+        action.type.split('/')[0..-2].map(&:camelize).join('::')
       end
     end
 
+    def action_type
```

Comment:
What do you think about adding the methods?:
```ruby
def subscribe?
  action.type == 'logux/subscribe'
end

def action?
   action.type != 'logux/subscribe'
end
```

Adding them would make the code more clear:

```ruby
raise Logux::NoActionError.new(message, meta: meta) if action?
```